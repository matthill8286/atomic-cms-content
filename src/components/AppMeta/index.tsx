import React, { useContext } from 'react';
import { Helmet } from 'react-helmet';
import {
  GraphCmsInstanceContext,
  InstanceContextTypes,
} from '@/context/instance';
import { AppMetaProps, AdditionalMeta } from './AppMeta.interface';

const MAX_LENGTH_OG_DESCRIPTION = 127;

export const AppMeta: React.FC<AppMetaProps> = ({
  title = '',
  description = '',
  generator = 'webmobile-landing-pages',
  keywords,
  canonical = '',
  imageUrl = '',
  prev,
  next,
  type = 'webpage',
  preconnectLinkList = [],
  prefetchLinkList = [],
  additionalMetas = [],
  includeProductInTitle = true,
  productName = '',
  children,
}) => {
  const { instance } = useContext<InstanceContextTypes>(
    GraphCmsInstanceContext
  );

  const additionalMeta: AdditionalMeta[] = additionalMetas.map(
    (meta, index) => ({
      key: index.toString(),
      ...meta,
    })
  );

  const isRobotsInAdditionalMeta = additionalMeta.find(
    item => item.name === 'robots'
  );

  !isRobotsInAdditionalMeta &&
    additionalMeta.push({ name: 'robots', content: 'follow' });

  const titleTemplate = includeProductInTitle ? `%s | ${productName}` : '%s';

  const meta: AdditionalMeta[] = [
    { name: 'description', content: description || '' },
    { name: 'keywords', content: [...(keywords || [])] },
    { name: 'generator', content: generator },
    { property: 'og:title', content: title || '' },
    {
      property: 'og:description',
      content: (description || '').slice(0, MAX_LENGTH_OG_DESCRIPTION).trim(),
    },
    { property: 'og:type', content: type },
    { property: 'og:image', content: imageUrl },
    { property: 'og:site_name', content: productName },
    { property: 'og:url', content: canonical },
    ...additionalMeta,
  ];

  const links = [
    ...(canonical ? [{ rel: 'canonical', href: canonical }] : []),
    ...(prev ? [{ rel: 'prev', href: prev }] : []),
    ...(next ? [{ rel: 'next', href: next }] : []),
    ...preconnectLinkList.map(link => ({ rel: 'preconnect', href: link })),
    ...prefetchLinkList.map(link => ({
      rel: 'preconnect dns-prefetch',
      href: link,
    })),
    ...(instance?.favicon
      ? [
          {
            rel: 'icon',
            href: instance?.favicon.url,
            type: 'image/png',
            sizes: '16x16',
          },
        ]
      : []),
  ];

  return (
    <Helmet
      titleTemplate={titleTemplate}
      defaultTitle={title}
      htmlAttributes={{ lang: 'en-GB' }}
      link={
        instance?.favicon
          ? [
              {
                rel: 'icon',
                type: 'image/png',
                href: instance?.favicon.url,
                sizes: '16x16',
              },
            ]
          : []
      }
    >
      {meta.map((entry, index) => {
        return <meta key={index} {...entry} />;
      })}
      {links.map((entry, index) => {
        return <link key={index} {...entry} />;
      })}
      <title>{title}</title>
      {children}
    </Helmet>
  );
};

export * from './AppMeta.interface';
