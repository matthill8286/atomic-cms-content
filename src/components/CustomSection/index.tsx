export * from './CustomSection.types'
export * from './CustomSection'
