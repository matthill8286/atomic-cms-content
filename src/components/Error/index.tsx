import React, { FC, useEffect } from 'react'
import { Button, Heading, Grid, Cell, Row } from '@matthill8286/atomic-ui'
import { useLocation } from 'react-router-dom'
import {
  StyledErrorPageContainer,
  StyledErrorCopyText,
  StyledErrorHeadingWrapper,
  StyledErrorImage,
  StyledErrorCode,
} from './ErrorPage.styled'
import { ErrorPageProps, ErrorPageType } from './ErrorPage.interface'
import { ConfigurableGraphCmsHtmlSerializer, Serializer } from '../Serializer'

export const IS_BROWSER = typeof window !== 'undefined'

export const ErrorPage: FC<ErrorPageProps> = ({
  errorType = ErrorPageType.TECHNICAL_ERROR,
  errorMessage,
}): JSX.Element => {
  const location = useLocation()
  const isOffline = IS_BROWSER && navigator?.onLine === false

  if (isOffline) {
    errorType = ErrorPageType.OFFLINE_ERROR
  }

  const isRedirected = location.pathname.includes('/error/500')

  useEffect(() => {
    if (isOffline) {
      window.addEventListener(
        'online',
        function() {
          window.location.reload()
        },
        false
      )
    }
  })

  interface ErrorMap {
    [key: string]: string | JSX.Element | undefined
  }

  const errorHeadline: ErrorMap = {
    [ErrorPageType.NOT_FOUND]: 'Not found',
    [ErrorPageType.TECHNICAL_ERROR]: 'Something went wrong',
    [ErrorPageType.OFFLINE_ERROR]: 'You are Offline',
  }

  const errorSubLabel: ErrorMap = {
    [ErrorPageType.NOT_FOUND]: 'This page has either been removed or is broken',
    [ErrorPageType.TECHNICAL_ERROR]: 'We are currently experiencing technical issues',
    [ErrorPageType.OFFLINE_ERROR]: 'Please connect to the internet',
  }

  const errorCodeLabel: ErrorMap = {
    [ErrorPageType.NOT_FOUND]: 'Error 404',
    [ErrorPageType.TECHNICAL_ERROR]: 'Error 500',
    [ErrorPageType.OFFLINE_ERROR]: 'Offline',
  }

  const buttonLabel: ErrorMap = {
    [ErrorPageType.NOT_FOUND]: 'Dashboard',
    [ErrorPageType.TECHNICAL_ERROR]: 'Reload',
    [ErrorPageType.OFFLINE_ERROR]: 'Offline Reload',
  }

  const svgImage: ErrorMap = {
    [ErrorPageType.NOT_FOUND]: null,
    [ErrorPageType.TECHNICAL_ERROR]: null,
    [ErrorPageType.OFFLINE_ERROR]: null,
  }

  const isTechnicalError = errorType === ErrorPageType.TECHNICAL_ERROR

  let buttonProps = {}
  if (errorType === ErrorPageType.NOT_FOUND || isRedirected) {
    buttonProps = {
      to: '/',
    }
  } else if (isTechnicalError) {
    buttonProps = {
      onClick: (event: any) => {
        event.preventDefault()
        window.location.reload()
      },
    }
  }

  const testLabels = {
    image: `error-page-image-${errorType}`,
    text: `error-page-text-${errorType}`,
  }

  const componentHeadings = (
    <>
      <StyledErrorHeadingWrapper>
        <Heading scale="level-2">{errorHeadline[errorType]}</Heading>
      </StyledErrorHeadingWrapper>
      <StyledErrorCopyText fontSize="sm" data-test={testLabels.text}>
        {errorSubLabel[errorType]}
      </StyledErrorCopyText>
    </>
  )

  return (
    <Grid>
      <Row>
        <Cell columns={12}>
          <StyledErrorPageContainer>
            <StyledErrorImage
              marginLeft={
                errorType === ErrorPageType.TECHNICAL_ERROR ||
                errorType === ErrorPageType.OFFLINE_ERROR
              }
              data-test={testLabels.image}>
              {svgImage[errorType]}
            </StyledErrorImage>
            {errorMessage?.message ? (
              <StyledErrorHeadingWrapper data-test={testLabels.text}>
                {Serializer(
                  errorMessage?.message,
                  // @ts-ignore
                  ConfigurableGraphCmsHtmlSerializer({})
                )}
              </StyledErrorHeadingWrapper>
            ) : (
              componentHeadings
            )}
            <Button size="lg" actionType="secondary" {...buttonProps}>
              {isRedirected ? 'Back to Home' : buttonLabel[errorType]}
            </Button>
            <StyledErrorCode scale="small">{errorCodeLabel[errorType]}</StyledErrorCode>
          </StyledErrorPageContainer>
        </Cell>
      </Row>
    </Grid>
  )
}

export default ErrorPage
export * from './ErrorPage.interface'
