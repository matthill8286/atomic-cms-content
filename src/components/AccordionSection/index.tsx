import * as React from 'react';
import {useLocation} from 'react-router';
import {Grid, Cell, Row, Offset, Accordion} from '@matthill8286/atomic-ui';
import {AccordionSectionProps} from './AccordionSection.types';
import {StyledAccordionWrapper} from './AccordionSection.styled';
import {CustomSection} from '../CustomSection';
import {Serializer, ConfigurableGraphCmsHtmlSerializer} from '../Serializer';

export * from './AccordionSection.types';

export const AccordionSection: React.FC<AccordionSectionProps> = ({
  entries = [],
  sectionColor,
  textColor = 'grey6',
  isLarge,
  withIconsOnRight,
  customAccordionEntries,
}) => {
  const location = useLocation();
  const anchoredAccordion = entries.find(
    entry => location.hash === `#${entry.anchorId}`
  );
  const openedIndex = anchoredAccordion && entries.indexOf(anchoredAccordion);

  const mappedAccordionEntries = entries.map(entry => {
    const accordionTitleComponent =
      entry.entryHeadline &&
      Serializer(
        entry.entryHeadline.raw,
        ConfigurableGraphCmsHtmlSerializer({
          fontSize: 'md',
          color: 'grey6',
          margins: 'xxs 0',
        })
      );

    return {
      details:
        entry.entryContent &&
        Serializer(
          entry.entryContent.raw,
          ConfigurableGraphCmsHtmlSerializer({
            primaryColor: sectionColor,
            inline: true,
            underline: false,
            margins: 'xxs 0',
          })
        ),
      title: accordionTitleComponent,
    };
  });

  return (
    <CustomSection color={sectionColor}>
      <Grid>
        <Row>
          <Offset colsXs={0} colsSm={0} columns={1} />
          <Cell colsXl={10} colsLg={10} colsMd={6} colsSm={8} colsXs={4}>
            <StyledAccordionWrapper textColor={textColor}>
              <Accordion
                isLarge={isLarge}
                withIconsOnRight={withIconsOnRight}
                customLabelHeading={{
                  bold: false,
                  margin: 'sm 0',
                  tag: 'div',
                }}
                labelPadding="xxs 0"
                entryPadding="0 0 sm"
                entries={mappedAccordionEntries}
                openedIndex={openedIndex}
              />
            </StyledAccordionWrapper>
          </Cell>
        </Row>
      </Grid>
    </CustomSection>
  );
};
