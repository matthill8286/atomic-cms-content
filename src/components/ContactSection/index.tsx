import React, {FC} from 'react';
import {
  Row,
  Cell,
  TwelveColumn,
  Picture,
  Heading,
  Offset,
  Grid,
} from '@matthill8286/atomic-ui';
import {ConfigurableGraphCmsHtmlSerializer, Serializer} from '../Serializer';

import {CustomSection} from '../CustomSection';
import {StyledPaddedSection} from './ContactSection.styled';
import {ContactSectionProps} from './ContactSection.types';
import {StyledTextSection} from '../TextSection/TextSection.styled';

export const ContactSection: FC<ContactSectionProps> = ({
  contactImage,
  contactHeading,
  richTextContactSection,
  sectionColor = 'clear',
  paddingTop,
  paddingBottom,
}) => {
  return (
    <CustomSection
      color={sectionColor}
      paddingTop={paddingTop}
      paddingBottom={paddingBottom}
    >
      <Grid>
        <Row noMargin>
          <Offset colsXl={1} colsLg={1} colsMd={0} colsSm={0} colsXs={0} />
          <Cell colsXl={10} colsLg={10} colsMd={8} colsSm={8} colsXs={4}>
            {contactHeading && (
              <Row noMargin>
                <Cell columns={12 as TwelveColumn}>
                  <Heading
                    scale="level-2"
                    tag="h2"
                    color="black"
                    margin="md 0"
                    padding="lg 0"
                    textAlign="center"
                  >
                    {contactHeading}
                  </Heading>
                </Cell>
              </Row>
            )}
            <Row noMargin>
              <Cell
                columns={6 as TwelveColumn}
                colsLg={6}
                colsMd={4}
                colsSm={8}
                colsXs={4}
              >
                <StyledPaddedSection
                  headingColor="black"
                  align="flex-start"
                  justify="center"
                  withPadding="false"
                >
                  {Serializer(
                    richTextContactSection.raw,
                    ConfigurableGraphCmsHtmlSerializer({
                      color: 'black',
                      margins: '0 0 xs 0',
                      inline: true,
                      forceNewTab: true,
                    })
                  )}
                </StyledPaddedSection>
              </Cell>
              <Cell
                columns={6 as TwelveColumn}
                colsLg={6}
                colsMd={4}
                colsSm={8}
                colsXs={4}
              >
                <StyledPaddedSection align="center" justify="center">
                  {contactImage?.url && (
                    <Picture
                      alt={contactImage?.altText}
                      src={contactImage?.url}
                      width="100%"
                      height="100%"
                    />
                  )}
                </StyledPaddedSection>
              </Cell>
            </Row>
          </Cell>
        </Row>
      </Grid>
    </CustomSection>
  );
};
