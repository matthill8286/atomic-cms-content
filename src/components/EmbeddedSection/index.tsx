import React, { FC } from 'react'
import { Grid, Row, Cell, Heading } from '@matthill8286/atomic-ui'
import { ConfigurableGraphCmsHtmlSerializer, Serializer } from '../Serializer'
import { CustomSection } from '../CustomSection'
import { StyledPaddedEmbedSection, StyledPaddedEmbed } from './EmbeddedSection.styled'
import { EmbeddedSectionProps } from './EmbeddedSection.types'

export const EmbeddedSection: FC<EmbeddedSectionProps> = ({
  embeddedSection,
  sectionColor,
  title,
}) => {
  return (
    <CustomSection color={sectionColor}>
      <Grid>
        <Row noMargin>
          <Cell columns={12}>
            <Heading scale="level-2" textAlign="center" margin="xxxxl 0 sm 0">
              {title}
            </Heading>
          </Cell>
        </Row>
        <Row noMargin>
          <Cell columns={12}>
            <StyledPaddedEmbedSection overrideOver>
              <StyledPaddedEmbed>
                {embeddedSection &&
                  Serializer(
                    embeddedSection.raw,
                    // @ts-ignore
                    ConfigurableGraphCmsHtmlSerializer({
                      textAlign: 'center',
                    })
                  )}
              </StyledPaddedEmbed>
            </StyledPaddedEmbedSection>
          </Cell>
        </Row>
      </Grid>
    </CustomSection>
  )
}
