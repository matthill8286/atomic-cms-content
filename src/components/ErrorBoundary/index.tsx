import React, { Component, ErrorInfo } from 'react'
import { ErrorPage as Error, ErrorPageType } from '../Error'
import { ErrorProps, ErrorState } from './ErrorBoundary.interface'

// Error boundaries need to be classes, therefore no FC
class ErrorBoundary extends Component<ErrorProps, ErrorState> {
  constructor(props: ErrorProps) {
    super(props)
    this.state = {
      error: undefined,
      errorInfo: undefined,
      errorTrackingObj: undefined,
    }
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    const errorTrackingObj = {
      errorTitle: error.message,
      errorCode: 500,
      requestUrl: window.location.pathname,
      requestedId: this.props.requestedId,
      pageType: this.props.pageType,
    }

    this.setState({ error, errorInfo, errorTrackingObj })
  }

  render() {
    const { Fallback } = this.props

    if (this.state.error) {
      if (Fallback) {
        return <Fallback />
      } else {
        return <Error errorType={ErrorPageType.TECHNICAL_ERROR} />
      }
    }

    return this.props.children
  }
}

export { ErrorBoundary }
