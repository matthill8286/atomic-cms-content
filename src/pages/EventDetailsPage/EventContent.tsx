import React from 'react';
import {CustomSection, FocusedSection, StaticPlaylist} from '../../components';

export const FOCUSED_SECTION = 'FocusedSection';

export const EventsWithAssetMeta = React.memo<any>(
  ({
    slices,
    dataTestId,
    sectionColor,
    overridePlaylistClickEvent,
    routeContext,
  }) => {
    return (
      <CustomSection color={sectionColor}>
        {slices?.map(
          (
            {
              pageComponents: {
                description,
                staticAssetTile,
                rightOrientation,
                staticPlaylists,
                profileSections,
                learningAssetMeta,
                richTextSection,
                heading,
                subHeading,
                id,
                name,
                __typename,
              },
              size,
            },
            index: number
          ) => {
            const keyProp = `${__typename}_${index}|${id}`;
            switch (__typename) {
              case 'EventFocusedArea':
                return (
                  <FocusedSection
                    key={keyProp}
                    title={richTextSection?.title}
                    description={richTextSection?.document}
                    profileBrandedLabel={subHeading}
                    profileTitle={heading}
                    learningAssetMeta={learningAssetMeta}
                    profileSections={profileSections}
                    sectionColor="grey1"
                    paddingTop="sm"
                    paddingBottom="sm"
                  />
                );
              case 'StaticPlaylistSection':
                if (!staticPlaylists.length) {
                  return null;
                }

                return (
                  <StaticPlaylist
                    key={keyProp}
                    playlistId={id}
                    dataTest={dataTestId}
                    staticPlaylists={staticPlaylists}
                    name={staticPlaylists?.[0].playlistName}
                    overridePlaylistClickEvent={overridePlaylistClickEvent}
                    routeContext={routeContext}
                    title={name}
                    playlistBackground={sectionColor ?? 'grey1'}
                    context="StaticPlaylist"
                    showLock
                    desktop={size ?? 4}
                    paddingTop="xxxl"
                    showDivider
                  />
                );
              default:
                return null;
            }
          }
        )}
      </CustomSection>
    );
  }
);
