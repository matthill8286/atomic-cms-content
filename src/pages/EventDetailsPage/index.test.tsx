import React from 'react';
import EventsDetailsPageWrapper from './index';
import {EventsWithAssetMeta} from './EventContent';
import {PageTemplateProps} from '../../components';

const props: PageTemplateProps = {
  landingPageContent: {
    slice: [],
    metaTitle: '',
    metaDescription: '',
    routeContext: 'events',
  },
  overridePlaylistClickEvent: false,
  withPageHead: true,
  PageContent: EventsWithAssetMeta,
  metaProps: {},
};

describe('EventDetailsPage', () => {
  it('renders correctly', () => {
    const wrapper = <EventsDetailsPageWrapper {...props} />;
    expect(wrapper).toMatchSnapshot();
  });
});
