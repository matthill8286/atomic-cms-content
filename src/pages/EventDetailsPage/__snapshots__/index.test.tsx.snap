// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetailsPage renders correctly 1`] = `
<EventsDetailsPageWrapper
  PageContent={
    Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "type": [Function],
    }
  }
  landingPageContent={
    Object {
      "metaDescription": "",
      "metaTitle": "",
      "routeContext": "events",
      "slice": Array [],
    }
  }
  metaProps={Object {}}
  overridePlaylistClickEvent={false}
  withPageHead={true}
/>
`;
